File: lingting-spring-security/src/main/java/live/lingting/spring/security/conditional/ConditionalOnUsingRemoteAuthorization.java
package live.lingting.spring.security.conditional;

import live.lingting.spring.security.configuration.SecurityResourceConfiguration;
import live.lingting.spring.security.properties.SecuritySpringProperties;
import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author lingting 2024-02-05 17:21
 */
@Target({ ElementType.TYPE, ElementType.METHOD })
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Inherited
@ConditionalOnBean(SecurityResourceConfiguration.class)
@ConditionalOnProperty(prefix = SecuritySpringProperties.PREFIX + ".authorization", value = "remote",
		havingValue = "true")
public @interface ConditionalOnUsingRemoteAuthorization {

}

File: lingting-spring-security/src/main/kotlin/live/lingting/spring/security/conditional/ConditionalOnUsingRemoteAuthorization.kt
package live.lingting.spring.security.conditional

import java.lang.annotation.Inherited
import live.lingting.spring.security.configuration.SecurityResourceConfiguration
import live.lingting.spring.security.properties.SecuritySpringProperties
import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty

/**
 * @author lingting 2024-02-05 17:21
 */
@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Inherited
@ConditionalOnBean(SecurityResourceConfiguration::class)
@ConditionalOnProperty(prefix = SecuritySpringProperties.PREFIX + ".authorization", value = ["remote"], havingValue = "true")
annotation class ConditionalOnUsingRemoteAuthorization
